Address static analyzer warnings in DefaultWebBrowserChecks.mm

rniwa opened the pull request:
#### b78c228354846fc6dd7df907f35d5de6b46ef306
<pre>
Address static analyzer warnings in DefaultWebBrowserChecks.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=287425">https://bugs.webkit.org/show_bug.cgi?id=287425</a>

Reviewed by Geoffrey Garen.

Addressed the static analyzer warnings in DefaultWebBrowserChecks.mm by protecting the queue as needed.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::itpQueue):
(WebKit::determineTrackingPreventionState):
(WebKit::doesAppHaveTrackingPreventionEnabled):

Canonical link: <a href="https://commits.webkit.org/290197@main">https://commits.webkit.org/290197@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b021f40d2542766a9a70dba552147c2f2d5323b7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89203 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8727 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44032 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94185 "Hash b021f40d for PR 40367 does not build (failure)") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39964 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16963 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/94185 "Hash b021f40d for PR 40367 does not build (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26402 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92205 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6989 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80961 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/94185 "Hash b021f40d for PR 40367 does not build (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6738 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39071 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77100 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36328 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96018 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16383 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12043 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77611 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16639 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76748 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76904 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21313 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19922 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9501 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13990 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16397 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21708 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16138 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19589 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17919 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
```diff
@@ -132,7 +132,7 @@ static bool determineTrackingPreventionStateInternal(bool appWasLinkedOnOrAfter,
 }
 
-static RefPtr<WorkQueue>& itpQueue()
+static RefPtr<WorkQueue>& itpQueueSingleton()
```
This object isn't really a singleton; we destroy and re-create it.

How about adding

```
static RefPtr<WorkQueue> protectedITPQueue()
{
    return itpQueue();
}
```

and changing call sites to use that?

geoffreygaren replied:
Or you can just do `Ref { itpQueue() }` at the call site.

rniwa replied:
Oh, duh, you're right. Adding protected version makes sense to me.

rniwa committed:
Address static analyzer warnings in DefaultWebBrowserChecks.mm
https://bugs.webkit.org/show_bug.cgi?id=287425

Reviewed by Geoffrey Garen.

Addressed the static analyzer warnings in DefaultWebBrowserChecks.mm by protecting the queue as needed.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::itpQueue):
(WebKit::determineTrackingPreventionState):
(WebKit::doesAppHaveTrackingPreventionEnabled):

Canonical link: https://commits.webkit.org/290197@main
